clients/java/src/main/java/io/camunda/zeebe/client/api/ExperimentalApi.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates a public API that can change at any time, and has no guarantee of API stability and
+ * backward-compatibility. If users want stabilization or signature change of a specific API that is
+ * currently annotated {@code @ExperimentalApi}, please comment on its tracking issue on github with
+ * rationale, usecase, and so forth, so that the Zeebe team may prioritize the process toward
+ * stabilization of the API.
+ *
+ * <p>Usage guidelines:
+ *
+ * <ol>
+ *   <li>This annotation is used only on public API. Internal interfaces should not use it.
+ *   <li>After Zeebe has gained API stability, this annotation can only be added to new API. Adding
+ *       it to an existing API is considered API-breaking.
+ *   <li>Removing this annotation from an API gives it stable status.
+ * </ol>
+ *
+ * <p>Note: This annotation is intended only for Zeebe library code. Users should not attach this
+ * annotation to their own code.
+ *
+ * <p>This annotation was originally copied from io.grpc.ExperimentalApi, licensed under the Apache
+ * License, Version 2.0. Copyright 2015 The gRPC Authors. Changes have been made since.
+ */
+@Retention(RetentionPolicy.CLASS)
```

Comment:
@korthout There is actually no advantage over the current one since both should work. It is only more accurate to use `RetentionPolicy.SOURCE` if we need `ExperimentalApi` annotation during compile time. Because `RetentionPolicy.CLASS` means you can read the annotation from the byte code. I am fine with keeping it as it is. 👍

Reply:
I see, thanks for clarifying @berkaycanbc. I'd like to leave it as `RetentionPolicy.CLASS`. This way, it's also visible at runtime through reflection. (💭 I  don't advocate that anyone should ever check this through reflection, but it might be expected).

Reply:
Yes, no harm for keeping it! An interesting note; you can only read the annotation using reflection when you use `RetentionPolicy.RUNTIME`. When you use `.CLASS`, the annotation is available in the byte code only. More interestingly, `.CLASS` is the default for retention type. 🤷‍♂️

Reply:
Oh TIL 😄 Thanks for sharing that

Reply:
`RetentionPolicy.CLASS` means the annotation is only visible at compile time though, no? My understanding is SOURCE is for annotations used during immediate compilation, CLASS for annotations which may be used during compilation of third party code which links against your compiled JAR (the one containing the annotated code), and RUNTIME for annotations which can be accessed reflectively. I think here CLASS is the appropriate one, unless we want users to be able to write ArchUnit rules or something like that. The grpc one is CLASS because they have an ErrorProne plugin which checks for usage of APIs annotated with it (which may be found in pre-compiled class files as well) and emits warnings/errors.

Reply:
>The grpc one is CLASS because they have an ErrorProne plugin which checks for usage of APIs annotated with it (which may be found in pre-compiled class files as well) and emits warnings/errors.

I removed that part of the docs because we don't have/need such a check

Reply:
@npepinpe Good one! Now, I think `.CLASS` makes more sense because it is possible that the annotation we introduced can be used for a similar purpose. 👍